zoom_monitor.py

```diff
@@ -0,0 +1,74 @@
+from contextlib import contextmanager
+import time
+
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver import Chrome
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+@contextmanager
+def monitor_zoom(url):
+    zoom = ZoomMonitor(url)
+    try:
+        yield zoom
+    finally:
+        zoom.clean_up()
+
+
+class ZoomMonitor():
+    def __init__(self, url):
+        self.driver = Chrome()
+
+        # The links that we receive prompt you to open the Zoom app if it's
+        # available. Replace the domain name to skip that.
+        updated_url = url.replace("viam.zoom.us/j", "app.zoom.us/wc/join")
+        self.driver.get(updated_url)
+
+        # Set our name and join the meeting
+        self.wait_for_element(By.ID, "input-for-name")
+        name_field = self.driver.find_element(By.ID, "input-for-name")
+        name_field.send_keys("Hand Raiser Bot")
+        self.driver.find_element(By.CSS_SELECTOR, ".zm-btn").click()
+
+        self.wait_for_element(By.CLASS_NAME, "SvgParticipantsDefault")
+        time.sleep(1) # The DOM isn't all set up yet; wait a little longer
```

Comment:
I really don't like this line, but can't figure out what else we're supposed to wait for instead. Any help here is appreciated!

Reply:
tested on another laptop, and the sleep was not enough. Will be changing it so that there is a for loop that tries a set number of times, and sleeps between each attempt.